src/ssl_load.c

```diff
@@ -2911,6 +2911,91 @@ static int LoadSystemCaCertsWindows(WOLFSSL_CTX* ctx, byte* loaded)
     return ret;
 }
 
+#ifdef WOLFSSL_WIN_USER_CA_CERTS
+static int LoadUserCaCertsWindows(WOLFSSL_CTX* ctx, const char* userStore,
+                                  const char* userDwFlags, const char* userPvPara) {
+    int ret = WOLFSSL_SUCCESS;
+    HANDLE handle = NULL;
+    PCCERT_CONTEXT certCtx = NULL;
+    LPCSTR storeProvider = (LPCSTR)userStore;
+    DWORD dwFlags;
+    const void* pvPara = userPvPara;
+
+    dwFlags = WinStrToDwFlags(userDwFlags);
+    if (!dwFlags) {
+        WOLFSSL_MSG("Invalid/unsupported dwFlags passed to LoadUserCaCertsWindows.");
+        return WOLFSSL_FAILURE;
+    }
+
+    handle = CertOpenStore(userStores, X509_ASN_ENCODING, NULL, dwFlags,
+                           userPvPara);
+    if (handle != NULL) {
+        while ((certCtx = CertEnumCertificatesInStore(handle, certCtx))
+                != NULL) {
+            if (certCtx->dwCertEncodingType == X509_ASN_ENCODING) {
+                if (ret = ProcessBuffer(ctx, certCtx->pbCertEncoded,
+                        certCtx->cbCertEncoded, WOLFSSL_FILETYPE_ASN1,
+                        CA_TYPE, NULL, NULL, 0,
+                        GET_VERIFY_SETTING_CTX(ctx)) != WOLFSSL_SUCCESS) {
+                    WOLFSSL_MSG_EX("Failed to process user cert: %d.", ret);
+                }
+            }
+        }
+    }
+    else {
+        WOLFSSL_MSG_EX("Failed to open user cert store.  Error: 0x%x.", GetLastError());
+        ret = WOLFSSL_FAILURE;
+    }
+
+    if (handle != NULL && !CertCloseStore(handle, 0)) {
+        WOLFSSL_MSG_EX("Failed to close user cert store.");
+        ret = WOLFSSL_FAILURE;
+    }
+
+    return ret;
+}
+
+DWORD WinStrToDwFlags(const char* dwFlags) {
+    if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_CURRENT_SERVICE")) {
+        return CERT_SYSTEM_STORE_CURRENT_SERVICE;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_CURRENT_USER")) {
+        return CERT_SYSTEM_STORE_CURRENT_USER;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_CURRENT_USER_GROUP_POLICY")) {
+        return CERT_SYSTEM_STORE_CURRENT_USER_GROUP_POLICY;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_LOCAL_MACHINE")) {
+        return CERT_SYSTEM_STORE_LOCAL_MACHINE;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_LOCAL_MACHINE_ENTERPRISE")) {
+        return CERT_SYSTEM_STORE_LOCAL_MACHINE_ENTERPRISE;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_LOCAL_MACHINE_GROUP_POLICY")) {
+        return CERT_SYSTEM_STORE_LOCAL_MACHINE_GROUP_POLICY;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_SERVICES")) {
+        return CERT_SYSTEM_STORE_SERVICES;
+    }
+    else if (!XSTRCMP(dwFlags, "CERT_SYSTEM_STORE_USERS")) {
+        return CERT_SYSTEM_STORE_USERS;
+    }
+
+    return NULL;
+}
+
+int wolfSSL_CTX_load_windows_user_CA_certs(WOLFSSL_CTX* ctx, const char* userStore,
```

Comment:
Any reason not to integrate this with `wolfSSL_CTX_load_system_CA_certs`? Is `wolfSSL_CTX_load_windows_user_CA_certs` a compatibility API? If not I would avoid making it Windows specific in case we wanted to expand its coverage beyond Windows.

Reply:
The customer has requested the ability to configure which cert store is loaded at runtime in wolfSSH, which requires arguments to be passed in to wolfSSL.  This is not an OpenSSL compatibility API, but it didn't make sense to me to add Windows specific arguments to wolfSSL_CTX_load_system_CA_certs.  I am not sure how to avoid adding a Windows specific API while still allowing wolfSSH to configure this at runtime.

Reply:
Seems like we could come up with a better name or API that would be easier to extend in the future. @ejohnstown I'd like your input as well on this since it is related to an SSH PR.

Reply:
Could there be a generic wolfSSL_CTX_load_CA_cert_store()? There can be a ctx pointer to a struct with OS specific data. For Windows, that would have a flag for user or system. Maybe at some point someone will want certs stored in a LDAP database. Perhaps the existing load function can be rewritten as a specific use case for this function.